src/option.rs

```diff
@@ -134,22 +135,21 @@ impl<T: crate::Serialize + crate::DeserializeOwned> Miniconf for core::option::O
             return Err(Error::PathAbsent);
         }
 
-        let (value, len) = serde_json_core::from_slice(value)?;
-        *self = Some(value);
-        Ok(len)
+        *self = Some(serde::Deserialize::deserialize(de).map_err(|_| Error::Deserialization)?);
```

Comment:
Hmm, it's occurring to me that we no longer handle a "data remains after deserialization" as an error. So technically, something like: `{"data": 5}blahblahblah` would still deserialize a `Map` properly, wouldn't it?

Reply:
That's handled in `Serde::set()` (the `de.end()`).